b1ddi/data_source_ipam_address_block_test.go

```diff
@@ -20,11 +20,18 @@ func TestAccDataSourceIpamsvcAddressBlock_Basic(t *testing.T) {
 							"name" = "tf_acc_test_address_block"
 						}
 					}
+					data "b1ddi_address_blocks" "tf_acc_address_blocks_by_tag" {
+						tfilters = {
+							# Search by Tag
+							"TestType" = "Acceptance"
+						}
```

Comment:
You should create an AB with this tag and should actually be checking `results.0.tags.TestType` = `Acceptance` and `results.0.tags.%` = `1`

What you are checking actually is the data block object itself which will always be ok

Please do the same for other objects